### DISCUSSION TITLE
Simulation stopping without known reason

### DISCUSSION THREAD

**Discussion by beltzjr** [JSBSim-Team/jsbsim#447] [Q&A]
New to JSBSim so any help is appreciated. Using the paraglider aircraft, I increased the IC for altitude to 100000, ubody to 900 and using the following script.

[Glider_High_Alt.txt](https://github.com/JSBSim-Team/jsbsim/files/6666956/Glider_High_Alt.txt)

After 1500s the simulation just stops. No reason as to why.

Same issue with an altitude of 500000 but after about 30s.

**Comment by seanmcleod** [comment]
I see the following, after setting IC for altitude to 100,000ft (ignoring how realistic this might be) and ubody = 900ft/s.

```sh
(base) C:\source\jsbsim>.\Release\JSBSim.exe --script=scripts\Glider_High_Alt.xml

....
....

Repeating Notify (Event 1) executed at time: 1930.007561
    position/h-sl-ft = 11150.111879
    position/h-agl-ft = 11150.111879
    velocities/vt-fps = 36.466855
    attitude/phi-rad = 0.219026
    attitude/theta-rad = 0.062440
    attitude/psi-rad = 3.927123


Repeating Notify (Event 1) executed at time: 1940.007557
    position/h-sl-ft = 10957.078374
    position/h-agl-ft = 10957.078374
    velocities/vt-fps = 36.355757
    attitude/phi-rad = 0.217500
    attitude/theta-rad = 0.057458
    attitude/psi-rad = 5.744115

0: GEAR_CONTACT: 1943.46 seconds: LMLG 1
1: GEAR_CONTACT: 1943.46 seconds: RMLG 1

Terminate (Event 2) executed at time: 1943.465889
    simulation/sim-time-sec = 1943.465889
```

So it terminates after 1943.5s when the gear makes contact with the ground. It does appear as if something goes 'wrong' just after 1940s, in the previous 10s it only descended ~150ft but in the next 3s it drops from 10,957ft to 0ft.

**Reply by beltzjr** [reply]
In FGLGear.cpp I printed out height after it was calculated (about line 295). When I was getting GEAR_CONTACT it was jumping to -1e6.

In my above case it just stops  but with height output it appears to suddenly climb 20,000ft.

34790: Height: 183230
34791: Height: 183229
34792: Height: 200116
34793: Height: 200117

Ran it another time at a different altitude and this is the final heights.

361745: Height: 31747.1
361746: Height: 31646.6
361747: Height: 31647.4
361748: Height: 6.67947e+06
361749: Height: 6.67947e+06

My assumption is there is something triggering height to be bad? Is earth assumed to just be round in terrain? Is there an easy way to debug in Visual Studio? Currently I am running from a terminal but compiling from VS.

**Comment by seanmcleod** [comment]
What does 34790 represent? The time in seconds? If so what has changed that the simulation doesn't end after 1500s as you originally reported?

**Reply by beltzjr** [reply]
It should be the buffer number.

ostringstream buf;
buf << "Height: " << height;
PutMessage(buf.str());

I have played with initial Altitude and velocity but the simulation keeps stopping around 31,000 ft.

**Comment by bcoconni** [comment]
I would say that an initial velocity of 900 ft/s is far too high: it corresponds to a Mach number of 0.93 which is unrealistic for a paraglider.

The rotational velocity r (rotational velocity around the paraglider Z axis - property `velocities/r-rad_sec`) reaches 30 rad/sec after a few seconds i.e. \~300 rpm. When the simulation time reaches ~1939 s, this results in the aerodynamic forces going through the roof and the paraglider is suddenly catapulted in orbit (literally :smile:).

I would suggest to choose wiser/more realistic initial conditions and make sure that they lead to a stable flight for the first few seconds of simulation before going to a long duration simulation.

**Reply by beltzjr** [reply]
1) I apologize now but I have run so many variants at this point it is hard to keep track of exactly what I did when. 

2) That being said, using the same glider variant... I have 5 different runs (70k|170, 200k|500, 200k|2000, 300k|500, 300k|700 where altitude in ft | fps in u ) and for all 5 they crash at about 31kft. I haven't looked into the atmosphere model yet but this is around the tropopause. I'm wondering if there is an issue with the model transitioning from the stratosphere to the troposphere.

3) Same issue occurs if I have an initial velocity of 0 fps.

**Reply by bcoconni** [reply]
> I haven't looked into the atmosphere model yet but this is around the tropopause. I'm wondering if there is an issue with the model transitioning from the stratosphere to the troposphere.

The transition from the stratosphere to the troposphere is occurring at 11 km (36089 ft) in the ISA 1976 model which we are using for the Earth atmosphere so this is unlikely to be correlated.
https://github.com/JSBSim-Team/jsbsim/blob/a35be5ce6f7d7ca0ef9a11a62563e54d769bc4f4/src/models/atmosphere/FGStandardAtmosphere.cpp#L84-L96

**Comment by seanmcleod** [comment]
Have you plotted the angular velocities for these 5 variants? As @bcoconni mentioned it appears as if the aircraft was in a flat spin at a ridiculous rate of ~300rpm. Maybe at these sort of rates and hitting a denser atmosphere some floating point numbers are blowing up.

Given your extreme starting altitudes, and your query about the Mars atmosphere model etc. I'm guessing you're trying to simulate some sort of Mars lander system entering from orbit? Trying to understand your starting requirements etc.

**Reply by beltzjr** [reply]
In general looking/investigating a glider from the upper atmosphere to the surface of various planets. For now just trying earth as it appears is all that's readily available. 

Just looking at rates now. It does look like rdot is getting out of control. I'll try changing different parameters and see if that helps.

![image](https://user-images.githubusercontent.com/85975672/122605537-32df2780-d03d-11eb-8f30-765929adeaba.png)

**Reply by bcoconni** [reply]
> In general looking/investigating a glider from the upper atmosphere to the surface of various planets.

Are you sure, you are using the right vehicle ? I mean a paraglider is nothing more than a person suspended to a parachute with a (very) small engine on his/her back:

![Paraglider image](https://upload.wikimedia.org/wikipedia/commons/thumb/8/82/Paraglider_landing.jpg/640px-Paraglider_landing.jpg)

If you want to experiment with a body decelerated by a parachute you may actually want to use the script [`ball_chute.xml`](https://github.com/JSBSim-Team/jsbsim/blob/master/scripts/ball_chute.xml) ?

**Reply by beltzjr** [reply]
Really want more of a sail plane. Paraglider seemed like the closest default aircraft. See that assumption is wrong now. Trying to design my own aircraft now.

**Comment by seanmcleod** [comment]
In your original script I don't see you explicitly trimming the aircraft on startup. Unless you trim it, or explicitly set all the initial conditions including the control positions such that the aircraft is in trim you don't have much hope of the aircraft not spinning out of control. And keeping it in trim as time goes on.

**Reply by beltzjr** [reply]
I'll give that a try. I have been using the paraglider_rotate_test.xml as a template (doesn't have a trim section).
